src/audio/module_adapter/module_adapter.c

```diff
@@ -1098,6 +1294,8 @@ int module_adapter_copy(struct comp_dev *dev)
 	comp_dbg(dev, "module_adapter_copy(): start");
 
 	struct processing_module *mod = comp_get_drvdata(dev);
+	if (mod->dev->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_DP)
+		return module_adapter_copy_dp_queues(dev);
```

Comment:
no, a module that is to be run as DP must use sink/src interface

Reply:
it may be in almost any order in module_adapter_copy
just one condition: DP should be before sink_source
in fact maybe it will be better to change it a bit: 

```
	if (IS_PROCESSING_MODE_AUDIO_STREAM(mod))
		return module_adapter_audio_stream_type_copy(dev);

	if (IS_PROCESSING_MODE_RAW_DATA(mod))
		return module_adapter_raw_data_type_copy(dev);

	if (IS_PROCESSING_MODE_SINK_SOURCE(mod)) {
		if (mod->dev->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_DP) {
			return module_adapter_copy_dp_queues(dev);
		} else {
			return module_adapter_sink_source_copy(dev);
		}
	}

	comp_err(dev, "module_adapter_copy(): unknown processing_data_type");
	return -EINVAL;
```